save-backend/src/main/kotlin/org/cqfn/save/backend/service/ProjectService.kt

```diff
@@ -9,4 +10,14 @@ import org.springframework.stereotype.Service
  * @property projectRepository
  */
 @Service
-class ProjectService(private val projectRepository: ProjectRepository)
+class ProjectService(private val projectRepository: ProjectRepository) {
+    /**
+     * @param project
+     */
+    fun saveProject(project: Project) {
+        projectRepository
+            .findAll()
```

Comment:
You can use [query by example](https://www.baeldung.com/spring-data-query-by-example) to retrieve only data of interest, and without filtering large amount of data in the service. You can also ignore id when defining the matcher, then custom `equals` will not be needed.